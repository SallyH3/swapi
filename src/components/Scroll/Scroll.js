import React from 'react';

export default function Scroll ({ crawl, title, year }) {
  return (
    <section className='scroll-container'>
      <section className='crawl'>
        <p className='text crawl'>{ crawl }</p>
        <h3 className='scroll-title'>{ title }</h3>
        <p className='text year'>{ year }</p>
      </section>
    </section>
  )
}