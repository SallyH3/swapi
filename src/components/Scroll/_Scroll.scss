.scroll-container {
  color: $scroll-color;
  display: flex;
  font-family: $scroll-font;
  height: 47rem;
  justify-content: center;
  margin-top: 1.2rem;
  perspective: 12.3rem;
}

.crawl {
  position: relative;
  transform-origin: 50% 100%;
  animation: scroll 70s;
  animation-iteration-count: 1;
}

.text {
  font-size: 3.5rem;
}

.scroll-title {
  font-size: 5.5rem;
}

@keyframes scroll {
  0% {
      top: 0;
      transform: rotateX(20deg) translateZ(0);
  }
  100% {
      top: -6000px;
      transform: rotateX(25deg) translateZ(-2500px);
  }
}