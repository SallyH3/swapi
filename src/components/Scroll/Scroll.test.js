import React from 'react';
import Scroll from './Scroll';
import { shallow } from 'enzyme';

describe('Scroll', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Scroll />
    )
  });
  it ('should match snapshot', () => {
    expect (wrapper).toMatchSnapshot();
  });
})