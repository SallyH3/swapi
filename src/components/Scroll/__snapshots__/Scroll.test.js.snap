// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll should match snapshot 1`] = `ShallowWrapper {}`;
